src/MessageHandler/ActivityPub/Inbox/LikeHandler.php

```diff
@@ -55,8 +55,10 @@ public function __invoke(LikeMessage $message): void
                 }
 
                 $actor = $this->activityPubManager->findActorOrCreate($message->payload['actor']);
-                // Check if actor and entity aren't empty
-                if (!empty($actor) && !empty($entity)) {
+                // Check if actor and activity aren't empty
+                // the entity can be resolved from activity if it has values
+                if (!empty($actor) && !empty($activity)) {
```

Comment:
`$activity` is never set?


Also `$entity` is fetch above already on line 52.

Reply:
in the old way the flow was like (lookup) -> `$activity` -> `$entity`, the null check was at entity although it's already nullable at activity, and due to the way the data from `$activity` is used to resolve the target entity (the `$this->entityManager->getRepository($activity['type'])->find((int) $activity['id'])` idiom) tha'll result in aforementioned error at the `getRepository()` part because `$activity` is null

with #597 that lookup idiom is wrapped and this change no longer applies, as it'll always try to create the entity objects.
still, is it really needed to fetch/create potentially the entire object chain just to undo a (dis)like that most likely will not even exists in our record?